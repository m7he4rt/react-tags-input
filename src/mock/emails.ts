export const emails: string[] = ["contato@rarolabs.com.br","nao-responda@rarolabs.com.br"]