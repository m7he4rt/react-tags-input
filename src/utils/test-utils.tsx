import React, { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { TagProvider } from '@/context/TagContext';

type CustomRenderProps = {} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement, 
  { ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <TagProvider>
      {ui}
    </TagProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }