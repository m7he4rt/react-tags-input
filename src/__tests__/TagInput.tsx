import React from 'react'
import TagsInput from '@/components/TagInput'
import userEvent from '@testing-library/user-event';
import { render, screen, cleanup, waitFor } from '@/utils/test-utils';

import { emails } from '@/mock/emails';

describe('TagsInput Component', () => {
  afterEach(jest.clearAllMocks)
  afterEach(cleanup)

  it('cria o snapshot do component', () => {
    const container = render(<TagsInput />)
    expect(container.asFragment()).toMatchSnapshot()
  })

  it('deve renderizar as tags enviadas por atributos', () => {
    render(<TagsInput tags={emails} />)
  })

  it('deve renderizar tags quando preencher o input e pressionar enter', async () => {
    render(<TagsInput />)
    const input = screen.getByRole('textbox')
    const text = "raro@gmail.com"

    userEvent.type(input, text);
    userEvent.keyboard('[Enter]');

    await waitFor(() => {
      expect(screen.getByRole('button', {name: 'raro@gmail.com'})).toBeInTheDocument();
    });
  })

  it('deve renderizar tags quando preencher o input e pressionar tab', async () => {
    render(<TagsInput />)
    const input = screen.getByRole('textbox')
    const text = "raro@gmail.com"

    userEvent.type(input, text);
    userEvent.tab();

    await waitFor(() => {
      expect(screen.getByRole('button', {name: 'raro@gmail.com'})).toBeInTheDocument();
    });
  })

  it('deve deletar a útima tag criada ao pressionar o botão de backspace', async () => {
    render(<TagsInput />)
    const input = screen.getByRole('textbox')
    const text = "raro@gmail.com;nao-responda@gmail.com"

    userEvent.type(input, text);
    userEvent.keyboard('[Enter]');

    expect(screen.getByRole('button', {name: 'raro@gmail.com'})).toBeInTheDocument();
    expect(screen.getByRole('button', {name: 'nao-responda@gmail.com'})).toBeInTheDocument();
 
    userEvent.keyboard('[Backspace]');
    
    await waitFor(() => {
      expect(screen.queryByText('nao-responda@gmail.com')).not.toBeInTheDocument();
    });
  })
})
