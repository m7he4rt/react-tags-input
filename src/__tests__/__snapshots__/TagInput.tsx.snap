// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsInput Component cria o snapshot do component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    >
      <div
        class="MuiBox-root MuiBox-root-1"
      />
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
